internal/docstore/doc.go

```diff
@@ -48,4 +48,44 @@
 // If a struct doesn't have a DocstoreRevision field, then the logic described above
 // won't apply to documents read and written with that struct. All writes with the
 // struct will succeed even if the document was changed since the last Get.
+//
+//
+// Queries
+//
+// Docstore supports querying within a collection. Call the Query method on
+// Collection to obtain a Query value, Then build your query by calling Query methods
+// like Where, Limit and so on. Finally, call the Get method on the query to execute it.
```

Comment:
Side note: should we even have a `Limit` method? I'm not sure it makes much sense in the context of an API that doesn't supported sorting, only returns results one at a time, and does client-side filtering.

Reply:
Limit is among other things a server-side optimization, even if the client API returns one result at a time.

You're right about sorting, but I'm pretty sure we can't leave the sort order undefined forever. (Working on gathering evidence for that.)

I'm going to leave it for now.